<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="yuqimiao" />

<meta name="date" content="2020-09-08" />

<title>Problem Intro and simulation setting</title>

<script src="site_libs/header-attrs-2.1/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multiomics-SIMLR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yuqimiao/multiomics-SIMLR">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Problem Intro and simulation setting</h1>
<h4 class="author">yuqimiao</h4>
<h4 class="date">2020-09-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>multiomics-SIMLR/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200830code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200830)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200830code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200830)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyuqimiaomultiomicsSIMLRtree7bea59cb6b646423ca74cbbc15f790406972a14etargetblank7bea59ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yuqimiao/multiomics-SIMLR/tree/7bea59cb6b646423ca74cbbc15f790406972a14e" target="_blank">7bea59c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyuqimiaomultiomicsSIMLRtree7bea59cb6b646423ca74cbbc15f790406972a14etargetblank7bea59ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/yuqimiao/multiomics-SIMLR/tree/7bea59cb6b646423ca74cbbc15f790406972a14e" target="_blank">7bea59c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/R/.DS_Store
    Ignored:    code/functions/.DS_Store
    Ignored:    code/functions/.Rhistory
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  SIMLR-tracking.md
    Untracked:  analysis/derivation_of_weights.jpeg
    Untracked:  analysis/gl_simlr_micro.Rmd
    Untracked:  code/MKerW-A/
    Untracked:  code/bck_scripts/
    Untracked:  code/cluster_functions/
    Untracked:  code/functions/Changing_rho.R
    Untracked:  code/functions/Code_5term_opt-copy.R
    Untracked:  code/functions/Code_5term_opt.R
    Untracked:  code/functions/SIMLR_no_weights.R
    Untracked:  code/functions/dist_kernels.R
    Untracked:  code/functions/gl_simlr.R
    Untracked:  code/functions/gl_simlr2.0.R
    Untracked:  code/functions/kernel_validation.R
    Untracked:  code/functions/microb_compare.R
    Untracked:  code/functions/simulation_function1.0.R
    Untracked:  code/functions/simulation_tidy.R
    Untracked:  code/functions/tmp.R
    Untracked:  code/microb_apply/
    Untracked:  code/simulation_1224.R
    Untracked:  code/simulation_1229.R
    Untracked:  code/simulation_compare2.R
    Untracked:  code/simulation_compare3.R
    Untracked:  code/simulation_compare_size.R
    Untracked:  data/balance_24_table.Rdata
    Untracked:  data/configuration_data_40.Rdata
    Untracked:  data/eff37_24_table.Rdata
    Untracked:  data/example_data/
    Untracked:  data/micro_case_pool.Rdata
    Untracked:  data/micro_case_pool_2.Rdata
    Untracked:  data/simu_1215/
    Untracked:  data/simu_1220/
    Untracked:  data/simu_1229/
    Untracked:  data/simu_all/
    Untracked:  data/simulation_2data/
    Untracked:  data/simulation_data_var10_s4/
    Untracked:  data/simulation_data_var20_s4/
    Untracked:  data/simulation_gl/
    Untracked:  data/simulation_gl_1/
    Untracked:  data/simulation_microgl/
    Untracked:  data/simulation_microgl_1/
    Untracked:  data/simulation_normalization/
    Untracked:  data/simulation_rho/
    Untracked:  data/simulation_var10_50/
    Untracked:  data/simulation_var5_50/
    Untracked:  data/table_balance_3d3g.Rdata
    Untracked:  gl-simlr_old_analysis.Rmd
    Untracked:  micro_abd.Rdata
    Untracked:  micro_abd_preabs.Rdata
    Untracked:  micro_case_pool2.Rdata
    Untracked:  microb_multisim.Rdata
    Untracked:  new_micro.jpeg
    Untracked:  parameter_separation/
    Untracked:  sub111_eff55_u20.rds
    Untracked:  tsne/

Unstaged changes:
    Deleted:    analysis/sim_report.rmd
    Modified:   code/R/utils.simlr.R
    Deleted:    code/functions/SNF_init_generater.R
    Deleted:    code/functions/kernel_buiding_functions.R
    Deleted:    code/functions/multiple.kernel.verify.R
    Modified:   code/functions/simulation_function.R
    Modified:   code/functions/simulation_verify.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/trivial_analysis.rmd</code>) and HTML (<code>docs/trivial_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yuqimiao/multiomics-SIMLR/blob/7bea59cb6b646423ca74cbbc15f790406972a14e/analysis/trivial_analysis.rmd" target="_blank">7bea59c</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-12-29
</td>
<td>
wflow_publish(files = c(“analysis/GL_SIMLR.rmd”, “analysis/trivial_analysis.rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yuqimiao/multiomics-SIMLR/d459d2bc8716de73ed4ed2f24ef82198649140ca/docs/trivial_analysis.html" target="_blank">d459d2b</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yuqimiao/multiomics-SIMLR/blob/449a753080d36403e17e8be8a82a87c9df19282a/analysis/trivial_analysis.rmd" target="_blank">449a753</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-23
</td>
<td>
update with new analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yuqimiao/multiomics-SIMLR/04bea8d1f8930683637af2b5d336323d0573f987/docs/trivial_analysis.html" target="_blank">04bea8d</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yuqimiao/multiomics-SIMLR/blob/43057b6dd55d50417e41963d275ea77bdfdc832b/analysis/trivial_analysis.rmd" target="_blank">43057b6</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-09
</td>
<td>
change small index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yuqimiao/multiomics-SIMLR/a03de3916debe2eed33606afd828bfe072a267f6/docs/trivial_analysis.html" target="_blank">a03de39</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yuqimiao/multiomics-SIMLR/blob/1179a63f49728013a527557cb74c091de4429f3e/analysis/trivial_analysis.rmd" target="_blank">1179a63</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-09
</td>
<td>
simulation describe and basic operation for a data in trivial analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(SNFtool)
library(rlist)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✓ ggplot2 3.3.0     ✓ purrr   0.3.3
✓ tibble  3.0.0     ✓ dplyr   0.8.5
✓ tidyr   1.0.2     ✓ stringr 1.4.0
✓ readr   1.3.1     ✓ forcats 0.5.0</code></pre>
<pre><code>── Conflicts ───────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(abSNF)</code></pre>
<pre><code>
Attaching package: &#39;abSNF&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:SNFtool&#39;:

    affinityMatrix, dist2, SNF, spectralClustering,
    standardNormalization</code></pre>
<pre class="r"><code>library(SIMLR)
library(parallel)
library(Matrix)</code></pre>
<pre><code>
Attaching package: &#39;Matrix&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:tidyr&#39;:

    expand, pack, unpack</code></pre>
<pre class="r"><code>source(&quot;./code/R/compute.multiple.kernel.R&quot;)
source(&quot;./code/functions/simulation_function.R&quot;)</code></pre>
<div id="goal-of-study" class="section level1">
<h1>Goal of study</h1>
<p>We want to find a data integration method which can accurately identify the subtypes embeded in multi-omics data of patients.</p>
<div id="problems-want-to-solve-in-this-method" class="section level2">
<h2>Problems want to solve in this method</h2>
<ol style="list-style-type: decimal">
<li>Find the structure of different omics data, and compare similarity/difference;</li>
<li>Integrate complementary information from multi-omics data to capture as much as subtype information as possible;</li>
<li>Avoid influence from noise features.</li>
</ol>
</div>
</div>
<div id="simulation-logic" class="section level1">
<h1>simulation logic</h1>
<div id="mechanism-and-characteristics-parameters" class="section level2">
<h2>Mechanism and characteristics parameters</h2>
<p>The goal of simulation is to embed differences in the signal features of different subtypes. Although for the same set of samples, they could imply different cluster information, and the difference can be shown from the effective size, the noise-to-signal ratio, the cluster type seperation and the feature distribution type.</p>
<p>Here we have 2 different settings to imply 2 scenarios:</p>
<p>Assuming 200 samples in 4 different subtypes, overall 4 data types. Every data type has 1000 features, with different noise-signal ratio and feature variance. The signal features are coming from different distributions with different parameters for different subtypes, which is the mechanism to generate subtype difference.</p>
<p>More specifically, let’s assume the first data type (n = 200 * p = 1000) can separate 4 subtypes with balanced subtype ratio(50 subjects per subtype). The first 50 features are the signals which actually effective in subtype separation. For the first subtype, the 50 feature signal are generated from <span class="math inline">\(N(\mu_1, \sigma^2)\)</span>, the second subtype, <span class="math inline">\(N(\mu_2, \sigma^2)\)</span> and so on, thus, the difference between <span class="math inline">\(\mu_i\)</span> and the size of <span class="math inline">\(\sigma\)</span> will influence the degree of separation of 4 subtypes. Also, not all features would follow a normal distribution. For example, beta measurement of methylation rates for a CpG sites could follow a beta distrbution.</p>
</div>
<div id="new-information-possession-setting" class="section level2">
<h2>[NEW] Information possession setting</h2>
<p>Besides the subtype ratio, <span class="math inline">\(\mu_i\)</span> and <span class="math inline">\(\sigma\)</span> and distribution type, there is another factor could influence the result of integration. Different data types could include different separation information. For example, subtype A and B have analygous gene expression profile, and subtype C and D also have analygous gene expression profile, but the subtype combinations A&amp;B and C&amp;D may have different gene expression profile. In this setting, using gene expression data only, we can separate A&amp;B and C&amp;D as 2 subtypes. Similarly, methylation data can separate A&amp;C and B&amp;D as 2 subtypes. In this setting, if we can properly integrate these 2 data types, we can have 4 subtype groups.But there are many possibilities for the information possession setting for each data type. Here we only consider 2 easy scenario.</p>
<ol style="list-style-type: decimal">
<li>different types of data imply same cluster information</li>
</ol>
<p>Here we assume all data types reflect same clustering information, which means all data types can separate 4 different subtypes with similar clustering analogy. We can assume for data type m, the signal features for 4 subtypes comes from <span class="math inline">\(N(\mu_{im}, \sigma^2), i = 1,2,3,4\)</span> separately. We set the parameter: <span class="math inline">\(\{\mu_{im}\} = \{-\mu_m,\mu_m,-2\mu_m,2\mu_m\}\)</span></p>
<pre class="r"><code>source(&quot;./code/functions/simulation_function.R&quot;)
sim1 = simulation_3(data_divide = rep(&quot;1/2/3/4&quot;,3),sigma = c(5,20,100))
mds_list = lapply(sim1[1:3], function(x){
  dist = dist2(x,x)
  mds = cmdscale(dist,2)
  return(mds)
})</code></pre>
<pre><code>Warning in if (is.na(c)) {: the condition has length &gt; 1 and only the first
element will be used

Warning in if (is.na(c)) {: the condition has length &gt; 1 and only the first
element will be used

Warning in if (is.na(c)) {: the condition has length &gt; 1 and only the first
element will be used</code></pre>
<pre class="r"><code>g_sim1 = as_tibble(list.rbind(mds_list)) %&gt;% 
  mutate(data_type = rep(c(1,2,3), each = 200),
         true = rep(sim1$truelabel, 3)) %&gt;% 
  ggplot(aes(x = V1, y = V2, color = factor(true)))+
  geom_point()+
  facet_grid(data_type~., scale = &quot;free&quot;)</code></pre>
<pre><code>Warning: The `x` argument of `as_tibble.matrix()` must have column names if `.name_repair` is omitted as of tibble 2.0.0.
Using compatibility `.name_repair`.
This warning is displayed once every 8 hours.
Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<pre class="r"><code>g_sim1</code></pre>
<p><img src="figure/trivial_analysis.rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yuqimiao/multiomics-SIMLR/blob/04bea8d1f8930683637af2b5d336323d0573f987/docs/figure/trivial_analysis.rmd/unnamed-chunk-2-1.png" target="_blank">04bea8d</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yuqimiao/multiomics-SIMLR/blob/a03de3916debe2eed33606afd828bfe072a267f6/docs/figure/trivial_analysis.rmd/unnamed-chunk-2-1.png" target="_blank">a03de39</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Note that in this setting, the distance between 4 clusters are different. Denoting distance between cluster 1 and 2 as <span class="math inline">\(d_{12}&#39;\)</span>, then we have <span class="math inline">\(d_{12}&#39;=d_{13}&#39; = d_{24}&#39;&lt;d_{14}&#39; = d_{23}&#39;&lt;d_{34}&#39;\)</span></p>
<ol start="2" style="list-style-type: decimal">
<li>different types of data imply different cluster information</li>
</ol>
<p>As mentioned above, in this setting, we assume 3 data types can separate 4 different subtypes, but different data types imply different clustering info. 3 data types separate AB/CD, AC/BD, AD/BC separately, with equall <span class="math inline">\((-\mu_1,\mu_1)\)</span> as mean of distribution uniformly.</p>
<p>Note that this is a completely balanced situation where all 4 subtypes are equally distant to each other.</p>
<pre class="r"><code>source(&quot;./code/functions/simulation_function.R&quot;)
sim2 = simulation_3(data_divide = c(&quot;12/34&quot;, &quot;13/24&quot;,&quot;14/23&quot;))
mds_list2 = lapply(sim2[1:3], function(x){
  dist = dist2(x,x)
  mds = cmdscale(dist,2)
  return(mds)
})</code></pre>
<pre><code>Warning in if (is.na(c)) {: the condition has length &gt; 1 and only the first
element will be used

Warning in if (is.na(c)) {: the condition has length &gt; 1 and only the first
element will be used

Warning in if (is.na(c)) {: the condition has length &gt; 1 and only the first
element will be used</code></pre>
<pre class="r"><code>g_sim2= as_tibble(list.rbind(mds_list2)) %&gt;% 
  mutate(data_type = rep(c(1,2,3), each = 200),
         true = rep(sim1$truelabel, 3)) %&gt;% 
  ggplot(aes(x = V1, y = V2, color = factor(true)))+
  geom_point()+
  facet_grid(data_type~., scale = &quot;free&quot;)

g_sim2</code></pre>
<p><img src="figure/trivial_analysis.rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yuqimiao/multiomics-SIMLR/blob/a03de3916debe2eed33606afd828bfe072a267f6/docs/figure/trivial_analysis.rmd/unnamed-chunk-3-1.png" target="_blank">a03de39</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="the-weight-optimization" class="section level1">
<h1>the weight optimization</h1>
<div id="intro-the-question" class="section level2">
<h2>intro the question</h2>
<p>From 50 running, we see that the mean weight for all 3 data types are balanced, but when looking at one-time output, there is always a dominant data, while the others contribute trivial. We need to figure out why this is the case.</p>
<pre class="r"><code># table_balance = NULL
# for(i in 1:10){
#   tmp = simulation_3(size = 150,sub_ratio = rep(1/3,3), eff_size = c(1,1,1), data_divide =
#                        c(&quot;1/23&quot;,&quot;12/3&quot;,&quot;2/13&quot;), sigma = 4)
#   res = simulation_verify_3data(K =3, tmp,tmp$truelabel)
#   table_balance = rbind(table_balance, res)
# }

load(&quot;./data/table_balance_3d3g.Rdata&quot;)
## show the direct output of weights for sym_balanced setting
table_balance[1:20,]%&gt;% select(contains(&quot;weight&quot;))</code></pre>
<pre><code># A tibble: 20 x 2
   weight1_sum weight2_sum
         &lt;dbl&gt;       &lt;dbl&gt;
 1      0.829       0.0680
 2      0.0888      0.812 
 3      0.0889      0.110 
 4      0.0954      0.0921
 5      0.0892      0.806 
 6      0.0895      0.108 
 7      0.0928      0.810 
 8      0.0791      0.106 
 9      0.102       0.0886
10      0.809       0.121 
11      0.120       0.795 
12      0.798       0.0953
13      0.0950      0.102 
14      0.105       0.0721
15      0.0943      0.805 
16      0.806       0.0859
17      0.805       0.0922
18      0.809       0.115 
19      0.806       0.0890
20      0.0916      0.800 </code></pre>
<pre class="r"><code>## the average of weight for sym_balanced setting
table_balance %&gt;% select(contains(&quot;weight&quot;) | contains(&quot;nmi&quot;)) %&gt;% summarise_all(mean)</code></pre>
<pre><code># A tibble: 1 x 7
  weight1_sum weight2_sum  nmi1  nmi2  nmi3 nmi_SNF nmi_SIMLR
        &lt;dbl&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;
1       0.348       0.342 0.497 0.545 0.504   0.992     0.758</code></pre>
</div>
<div id="derivation-of-weight" class="section level2">
<h2>Derivation of weight</h2>
<p>When fixing the similarity matrix</p>
<div class="figure">
<img src="derivation_of_weights.jpeg" alt="" />
<p class="caption">derivation of weight</p>
</div>
<p>From the derivation, we can see that the weight for a certain kernel depends on 2 parts:</p>
<ul>
<li>the sum of multiplication of distance and similarity matrix</li>
<li>the hyperparameter <span class="math inline">\(\rho\)</span></li>
</ul>
<div id="the-influence-of-rho" class="section level3">
<h3>the influence of rho</h3>
<p>Simulation of different scenario to see the effect of U</p>
<p>Scenario: * eff_size: c(5,5),c(3,7) * sub_ratio: rep(1/3,3),c(0.3,0.4,0.3)</p>
<p>Under every scenario, we test U = c(20,30,40,50) separately</p>
<p>We will focus * the sum of weight for first data type * the final performance.</p>
<pre class="r"><code>files = list.files(&quot;./data/simulation_rho/&quot;)
table_rho = NULL
for(i in 1:length(files)){
  table_cur = readRDS(paste(&quot;./data/simulation_rho/&quot;,files[i],sep = &quot;&quot;))
  table_rho = rbind(table_rho, table_cur)
}
## show the mean performance
table_rho_mean = table_rho %&gt;% 
  group_by(scenario,cluster) %&gt;% 
  summarise_all(mean) %&gt;% 
  mutate(nmi_improve = nmi_SIMLR-nmi_SNF) %&gt;% 
  arrange(desc(nmi_improve))

table_rho_mean%&gt;% 
  separate(scenario, into = c(&quot;sub_eff&quot;,&quot;U&quot;), sep = &quot;_(?=[^_]+$)&quot;) %&gt;%
  pivot_longer(
    nmi_SNF:nmi_SIMLR,
    names_to = &quot;methods&quot;,
    values_to = &quot;value&quot;
  ) %&gt;% 
  ggplot(aes(x = U,y = value,color = methods)) +
  geom_boxplot()+
  facet_grid(sub_eff~.)</code></pre>
<p><img src="figure/trivial_analysis.rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## show the weight change among U
table_rho %&gt;% 
  separate(scenario, into = c(&quot;sub_eff&quot;,&quot;U&quot;), sep = &quot;_(?=[^_]+$)&quot;) %&gt;%
  ggplot(aes(x = U,y = data1_weight,color = sub_eff)) +
  geom_boxplot()</code></pre>
<p><img src="figure/trivial_analysis.rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>From the simulation, we can see, when the variance is not extremely large, the performance of SIMLR will be relative stable. Only by changing weight, the performance of SIMLR can’t show greater improve than SNF</p>
<p>Here, we can see the parameter <span class="math inline">\(\rho\)</span> (Or in this algorithm U) has a dominant role to control the variation of weights between kernels and data types. In this algorithm, the U is set to be 20 for all situation, which may not be proper for all situation.</p>
</div>
</div>
<div id="effect-of-simlr" class="section level2">
<h2>Effect of SIMLR</h2>
<p>– see the change of integrated similarity matrix</p>
<pre class="r"><code>## Data simulation
set.seed(34)
sim_eff37 = simulation_2(eff_size = c(3,7),sigma = c(18,(49*2)))
## show the data
par(mfrow = c(1,1))
### data 2 has larger signals but also larger var
plot(1:150, rowMeans(sim_eff37$data2[,1:50]),col = &quot;red&quot;) 
### data1 has weaker signals also smaller var
points(1:150, rowMeans(sim_eff37$data1[,1:50]))</code></pre>
<p><img src="figure/trivial_analysis.rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## normalize data
data1 = standardNormalization(sim_eff37$data1)
data2 = standardNormalization(sim_eff37$data2)
truelabel = sim_eff37$truelabel
## fit SIMLR and get optimal weights separately
res1 = SIMLR(t(data1),c = 2)</code></pre>
<pre><code>Computing the multiple Kernels.
Performing network diffiusion.
Iteration:  1 
Iteration:  2 
Iteration:  3 
Iteration:  4 
Iteration:  5 
Iteration:  6 
Iteration:  7 
Iteration:  8 
Iteration:  9 
Iteration:  10 </code></pre>
<pre><code>Warning in SIMLR(t(data1), c = 2): Maybe you should set a larger value of c.</code></pre>
<pre><code>Performing t-SNE.
Epoch: Iteration # 100  error is:  1.362264 
Epoch: Iteration # 200  error is:  0.8925151 
Epoch: Iteration # 300  error is:  0.7063483 
Epoch: Iteration # 400  error is:  0.4523206 
Epoch: Iteration # 500  error is:  0.1335415 
Epoch: Iteration # 600  error is:  0.1335126 
Epoch: Iteration # 700  error is:  0.1335126 
Epoch: Iteration # 800  error is:  0.1335126 
Epoch: Iteration # 900  error is:  0.1335126 
Epoch: Iteration # 1000  error is:  0.1335126 
Performing Kmeans.
Performing t-SNE.
Epoch: Iteration # 100  error is:  15.01244 
Epoch: Iteration # 200  error is:  1.298033 
Epoch: Iteration # 300  error is:  0.97185 
Epoch: Iteration # 400  error is:  0.8181788 
Epoch: Iteration # 500  error is:  0.9069607 
Epoch: Iteration # 600  error is:  0.8901111 
Epoch: Iteration # 700  error is:  0.7958121 
Epoch: Iteration # 800  error is:  0.6989308 
Epoch: Iteration # 900  error is:  0.7612859 
Epoch: Iteration # 1000  error is:  0.1387673 </code></pre>
<pre class="r"><code>res2 = SIMLR(t(data2),c = 2)</code></pre>
<pre><code>Computing the multiple Kernels.
Performing network diffiusion.
Iteration:  1 
Iteration:  2 
Iteration:  3 
Iteration:  4 
Iteration:  5 
Iteration:  6 
Iteration:  7 
Iteration:  8 
Iteration:  9 
Iteration:  10 </code></pre>
<pre><code>Warning in SIMLR(t(data2), c = 2): Maybe you should set a larger value of c.</code></pre>
<pre><code>Performing t-SNE.
Epoch: Iteration # 100  error is:  1.194049 
Epoch: Iteration # 200  error is:  1.024211 
Epoch: Iteration # 300  error is:  1.0368 
Epoch: Iteration # 400  error is:  0.8314913 
Epoch: Iteration # 500  error is:  0.8539317 
Epoch: Iteration # 600  error is:  0.6738982 
Epoch: Iteration # 700  error is:  0.8938559 
Epoch: Iteration # 800  error is:  0.6988512 
Epoch: Iteration # 900  error is:  0.6376507 
Epoch: Iteration # 1000  error is:  0.2398244 
Performing Kmeans.
Performing t-SNE.
Epoch: Iteration # 100  error is:  16.03722 
Epoch: Iteration # 200  error is:  1.382657 
Epoch: Iteration # 300  error is:  1.001071 
Epoch: Iteration # 400  error is:  0.8125381 
Epoch: Iteration # 500  error is:  0.6905728 
Epoch: Iteration # 600  error is:  0.6427157 
Epoch: Iteration # 700  error is:  0.1741059 
Epoch: Iteration # 800  error is:  0.1650264 
Epoch: Iteration # 900  error is:  0.1649004 
Epoch: Iteration # 1000  error is:  0.1647973 </code></pre>
<pre class="r"><code>opt_w1 = res1$alphaK
opt_w2 = res2$alphaK
weight_list = list(opt_w1 = opt_w1, opt_w2 = opt_w2)

## get the combined multiple kernel separately
mk1 = multiple.kernel(data1,cores.ratio = 0.5)
mk2 = multiple.kernel(data2,cores.ratio = 0.5)
D_kernels = list()
for(i in 1:110){
  if(i&lt;=55){
    D_kernels[[i]] = mk1[[i]]
  }else if (i&lt;=110){
    D_kernels[[i]] = mk2[[i-55]]
  }
}</code></pre>
<div id="what-is-the-difference-btw-initial-s_0-and-s-after-simlr-optimization" class="section level3">
<h3>What is the difference btw initial S_0 and S after SIMLR optimization?</h3>
<pre class="r"><code>## we fisrt see the effect of averaged S_ij
### calculate the initial averaged S_ij
sum_kernels = matrix(0,150,150)
for(i in 1:length(D_kernels)){
  sum_kernels = sum_kernels+as.matrix(D_kernels[[i]])
}
s0 =  sum_kernels/length(D_kernels)
s0 = max(s0) - s0
cluster_s0 = spectralClustering(s0,3)
heatmap(s0, Rowv = NA, Colv = NA, scale = &quot;column&quot;)</code></pre>
<p><img src="figure/trivial_analysis.rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yuqimiao/multiomics-SIMLR/blob/d459d2bc8716de73ed4ed2f24ef82198649140ca/docs/figure/trivial_analysis.rmd/unnamed-chunk-8-1.png" target="_blank">d459d2b</a>
</td>
<td>
yuqimiao
</td>
<td>
2020-09-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## Now we see the similarity matrix resulting from SIMLR
source(&quot;./code/functions/SIMLR_multi.R&quot;)</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre><code>
Attaching package: &#39;igraph&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    as_data_frame, groups, union</code></pre>
<pre><code>The following objects are masked from &#39;package:purrr&#39;:

    compose, simplify</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    crossing</code></pre>
<pre><code>The following object is masked from &#39;package:tibble&#39;:

    as_data_frame</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    decompose, spectrum</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    union</code></pre>
<pre><code>Loading required package: cluster</code></pre>
<pre><code>
Attaching package: &#39;clValid&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:igraph&#39;:

    clusters</code></pre>
<pre class="r"><code>sim_res = SIMLR_multi(D_Kernels = D_kernels, c = 3)</code></pre>
<pre><code>Iteration:  1 
Iteration:  2 
Iteration:  3 
Iteration:  4 
Iteration:  5 
Iteration:  6 
Iteration:  7 
Iteration:  8 
Iteration:  9 
Iteration:  10 
Iteration:  11 
Iteration:  12 
Iteration:  13 
Iteration:  14 
Iteration:  15 
Iteration:  16 
Iteration:  17 
Iteration:  18 
Iteration:  19 
Iteration:  20 
Iteration:  21 
Iteration:  22 
Iteration:  23 
Iteration:  24 
Iteration:  25 
Iteration:  26 
Iteration:  27 
Iteration:  28 
Iteration:  29 
Iteration:  30 
[1] 150
Performing t-SNE.
Epoch: Iteration # 100  error is:  1.344411 
Epoch: Iteration # 200  error is:  1.225754 
Epoch: Iteration # 300  error is:  1.211949 
Epoch: Iteration # 400  error is:  1.207015 
Epoch: Iteration # 500  error is:  1.206727 
Epoch: Iteration # 600  error is:  1.206622 
Epoch: Iteration # 700  error is:  1.202785 
Epoch: Iteration # 800  error is:  1.202686 
Epoch: Iteration # 900  error is:  1.202584 
Epoch: Iteration # 1000  error is:  1.202486 
Performing Kmeans.
Performing t-SNE.
Epoch: Iteration # 100  error is:  20.97345 
Epoch: Iteration # 200  error is:  2.970315 
Epoch: Iteration # 300  error is:  2.228818 
Epoch: Iteration # 400  error is:  2.002058 
Epoch: Iteration # 500  error is:  1.771497 
Epoch: Iteration # 600  error is:  1.727241 
Epoch: Iteration # 700  error is:  1.662791 
Epoch: Iteration # 800  error is:  1.575334 
Epoch: Iteration # 900  error is:  1.59419 
Epoch: Iteration # 1000  error is:  1.588558 </code></pre>
<pre class="r"><code>s_res = sim_res$S
heatmap(s_res, Rowv = NA, Colv = NA, scale = &quot;column&quot;)</code></pre>
<p><img src="figure/trivial_analysis.rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>compare(sim_res$y$cluster,rep(c(1,2,3), each = 50), method = &quot;nmi&quot;)</code></pre>
<pre><code>[1] 0.6643379</code></pre>
<pre class="r"><code>compare(cluster_s0,rep(c(1,2,3), each = 50), method = &quot;nmi&quot;)</code></pre>
<pre><code>[1] 0.8894835</code></pre>
<p>As shown here, the multiplication sum has trivial difference for the initial <span class="math inline">\(S_0\)</span>, but after the optimization by SIMLR, the difference btw 2 multiplication_sum become larger.</p>
<p><em>Is this right? What is the role of S here? should the data be far away?</em></p>
</div>
</div>
</div>
<div id="possible-problem-and-innovation-of-simlr" class="section level1">
<h1>Possible problem and innovation of SIMLR</h1>
<div id="problem-of-simlr-in-this-simulation" class="section level2">
<h2>Problem of SIMLR in this simulation</h2>
<div id="different-clusters-has-different-contribution-requirment-from-data" class="section level3">
<h3>Different clusters has different contribution requirment from data</h3>
<div id="data-simulation" class="section level4">
<h4>Data simulation</h4>
<pre class="r"><code>set.seed(234)
## get 2 balanced data type
library(tidyverse)
source(&quot;./code/functions/simulation_function.R&quot;)
set.seed(111)
sim = simulation_2(size = 150, sub_ratio = rep(1/3,3), eff_size = c(5,5), sigma = c(100,100),data_divide = c(&quot;1/23&quot;,&quot;12/3&quot;),dist = c(&quot;normal&quot;,&quot;normal&quot;))</code></pre>
<pre class="r"><code>## separately fit the similarity matrix
res1 = SIMLR(t(sim$data1),2)
res2 = SIMLR(t(sim$data2),2)

## show the similarity difference
dist_s = dist2(res1$S, res2$S)
plot(density(dist_s))

plot(density(diag(dist_s)))

## show the dissimilar person within 2 similarity matrix
## use the last 25 quantiles of diagnal similarity
dissim = (1:150)[diag(dist_s) &gt; quantile(diag(dist_s))[4]]</code></pre>
</div>
</div>
</div>
<div id="possible-solutions" class="section level2">
<h2>Possible solutions</h2>
<div id="the-order-of-similarity-in-separate-simlr-results" class="section level3">
<h3>the order of similarity in separate SIMLR results</h3>
<p>Since the 2 data are seeparating different clusters, <em>the order of similarity</em> btw pairs should reflect some information:</p>
<p>For a given pair: * if 2 data both contain compliant info, i.e., both cluster them into 1, or both not, then th order should be similar * If 2 data contain different info, i.e., one data clustered the pair into 1 cluter, the other data clustered them separately, then the order should show difference.</p>
<p>Here we use a large variance balance case to verify the guess</p>
<pre class="r"><code>library(tidyverse)
source(&quot;./code/functions/simulation_function.R&quot;)
set.seed(111)
sim = simulation_2(size = 150, sub_ratio = rep(1/3,3), eff_size = c(5,5), sigma = c(100,100),data_divide = c(&quot;1/23&quot;,&quot;12/3&quot;),dist = c(&quot;normal&quot;,&quot;normal&quot;))

## see the distribution of 2 data types
par(mfrow = c(2,1))
plot(1:150, rowMeans(sim$data1[,1:50]),main = &quot;data1&quot;)
plot(1:150, rowMeans(sim$data2[,1:50]),main = &quot;data2&quot;)

## Do SIMLE separately
res1 = SIMLR(t(sim$data1),2)
res2 = SIMLR(t(sim$data2),2)

## see the difference of order of 2 similarity matrix
# sim_ord1 = sort(res1$S,index.return = T)$ix
# sim_ord2 = sort(res2$S,index.return = T)$ix
# dif_ord = match(sim_ord1,sim_ord2)-(1:length(sim_ord1))
# par(mfrow = c(1,1))
# plot(density(dif_ord)) 

# pair_to_Sij plot for both data
par(mfrow = c(2,1))
plot(1:150^2,as.vector(res1$S))
plot(1:150^2,res2$S)</code></pre>
<div id="what-if-we-directly-choose-the-largest-distance-in-2-data-for-every-pair" class="section level4">
<h4>What if we directly choose the largest distance in 2 data for every pair?</h4>
<pre class="r"><code>## first calculate the weighted distance
mk1 = multiple.kernel(sim$data1,cores.ratio = 0.5)
mk2 = multiple.kernel(sim$data2,cores.ratio = 0.5)

opt_dist1 = matrix(0,150,150)
for(i in 1:55){
  opt_dist1 = opt_dist1 + res1$alphaK[i]*mk1[[i]]
}

opt_dist2 = matrix(0,150,150)
for(i in 1:55){
  opt_dist2 = opt_dist2 + res2$alphaK[i]*mk2[[i]]
}
## then, for the similar ordered pairs, we use the min distance, for the dissimilar pairs, we use the max distance 
c1 = 1000
c2 = 1000
distX = matrix(0,150,150)
for(i in 1:150^2){
  if(dif_ord[i]&lt;-c1 |dif_ord[i] &gt; c2){
    distX[i] = max(opt_dist1[i],opt_dist2[i])
  }else{
    distX[i] = min(opt_dist1[i],opt_dist2[i])
  }
}
heatmap(distX,Rowv = NA, Colv = NA, scale = &quot;column&quot;)


## then we should change the SIMLR process, we don&#39;t need weight and multiple kernels, we only need the process of rank constraint and max multiplication_sum
source(&quot;./code/functions/SIMLR_no_weights.R&quot;)
res_sep = SIMLR_no_weight(distX,c = 3)
compare(res_sep$y$cluster, sim$truelabel,method = &quot;nmi&quot;)</code></pre>
</div>
</div>
<div id="to-think-add-freedom-of-weight-specific-weight-for-every-cluster" class="section level3">
<h3>[TO THINK] Add freedom of weight – specific weight for every cluster</h3>
<p>New term in the objective function: <span class="math display">\[\min_{w} \sum_{i,j,t,l}w_{tl}D_{l,i,j}S_{ij}I\{sub_{ij}\in C_t\}\]</span> t is for different cluster</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] clValid_0.6-9   cluster_2.1.0   igraph_1.2.5    gplots_3.0.3   
 [5] Matrix_1.2-18   SIMLR_1.12.0    abSNF_1.0.0     forcats_0.5.0  
 [9] stringr_1.4.0   dplyr_0.8.5     purrr_0.3.3     readr_1.3.1    
[13] tidyr_1.0.2     tibble_3.0.0    ggplot2_3.3.0   tidyverse_1.3.0
[17] rlist_0.4.6.1   SNFtool_2.3.0   workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] nlme_3.1-145       bitops_1.0-6       fs_1.4.1           lubridate_1.7.8   
 [5] RcppAnnoy_0.0.16   httr_1.4.1         rprojroot_1.3-2    tools_3.6.2       
 [9] backports_1.2.0    utf8_1.1.4         R6_2.4.1           KernSmooth_2.23-16
[13] DBI_1.1.0          colorspace_1.4-1   withr_2.1.2        tidyselect_1.0.0  
[17] compiler_3.6.2     git2r_0.26.1       cli_2.0.2          rvest_0.3.5       
[21] xml2_1.3.0         labeling_0.3       caTools_1.18.0     scales_1.1.0      
[25] digest_0.6.25      rmarkdown_2.1      pkgconfig_2.0.3    htmltools_0.4.0   
[29] prettyGraphs_2.1.6 dbplyr_1.4.2       rlang_0.4.8        readxl_1.3.1      
[33] rstudioapi_0.11    farver_2.0.3       generics_0.0.2     jsonlite_1.6.1    
[37] gtools_3.8.2       magrittr_1.5       heatmap.plus_1.3   Rcpp_1.0.5        
[41] munsell_0.5.0      fansi_0.4.1        lifecycle_0.2.0    stringi_1.4.6     
[45] whisker_0.4        yaml_2.2.1         grid_3.6.2         gdata_2.18.0      
[49] promises_1.1.0     crayon_1.3.4       lattice_0.20-41    haven_2.3.1       
[53] hms_0.5.3          knitr_1.28         pillar_1.4.3       codetools_0.2-16  
[57] reprex_0.3.0       glue_1.4.0         evaluate_0.14      alluvial_0.1-2    
[61] data.table_1.12.8  modelr_0.1.6       vctrs_0.3.4        httpuv_1.5.2      
[65] cellranger_1.1.0   gtable_0.3.0       ExPosition_2.8.23  assertthat_0.2.1  
[69] xfun_0.12          broom_0.5.5        pracma_2.2.9       RSpectra_0.16-0   
[73] later_1.0.0        class_7.3-16       ellipsis_0.3.0    </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
